crates/ruff/src/rules/refurb/rules/unnecessary_enumerate.rs

```diff
@@ -0,0 +1,311 @@
+use num_traits::ToPrimitive;
+use std::fmt;
+
+use crate::autofix::snippet::SourceCodeSnippet;
+use ruff_diagnostics::{AutofixKind, Diagnostic, Edit, Fix, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast as ast;
+use ruff_python_ast::{Arguments, Constant, Expr};
+use ruff_python_codegen::Generator;
+use ruff_python_semantic::helpers::is_unused;
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::checkers::ast::Checker;
+use crate::registry::AsRule;
+
+/// ## What it does
+/// Checks for uses of `enumerate` that discard either the index or the value
+/// when iterating over a sequence.
+///
+/// ## Why is this bad?
+/// The built-in `enumerate` function is useful when you need both the index and
+/// value of a sequence.
+///
+/// If you only need the index or values of a sequence, you should iterate over
+/// `range(len(...))` or the sequence itself, respectively, instead. This is
+/// more efficient and communicates the intent of the code more clearly.
+///
+/// ## Example
+/// ```python
+/// for index, value in enumerate(sequence):
+///     print(index)
+///
+/// for index, value in enumerate(sequence):
+///     print(value)
+/// ```
```

Comment:
I believe you meant this:

```suggestion
/// ```python
/// for index, _ in enumerate(sequence):
///     print(index)
///
/// for _, value in enumerate(sequence):
///     print(value)
/// ```
```

Reply:
The original intent was to demonstrate that the logic does more than check for underscores, but I think committing your suggestion makes the rule's intent more explicit (more important in documentation).